test/unittests/test_analyse_xml_data/test_extract_links_from_XML.py

```diff
@@ -0,0 +1,142 @@
+        '<resource xmlns="https://dasch.swiss/schema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" '
+        'label="res_A_11" restype=":TestThing" id="res_A_11" permissions="res-default"><text-prop '
+        'name=":hasRichtext"><text permissions="prop-default" encoding="xml"><a class="salsah-link" '
+        'href="IRI:res_B_11:IRI">res_B_11</a></text></text-prop><resptr-prop name=":hasResource1"><resptr '
+        'permissions="prop-default">res_B_11</resptr></resptr-prop></resource>'
+    )
+    res_resptr, res_xml = _get_all_links_from_one_resource(test_ele)
+    expected_resptr, expected_xml = ["res_B_11"], [{"res_B_11"}]
+    assert expected_resptr == res_resptr
+    assert expected_xml == unordered(res_xml)
+
+
+def test_get_all_links_from_one_resource_no_links() -> None:
+    test_ele = etree.fromstring(
+        '<resource label="res_B_18" restype=":TestThing" id="res_B_18" permissions="res-default"/>'
+    )
+    res = _get_all_links_from_one_resource(test_ele)
+    assert ([], []) == res
+
+
+def test_text_only_get_all_links_from_one_resource() -> None:
+    test_ele = etree.fromstring(
+        '<resource xmlns="https://dasch.swiss/schema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" '
+        'label="res_C_18" restype=":TestThing" id="res_C_18" permissions="res-default"><text-prop '
+        'name=":hasRichtext"><text permissions="prop-default" encoding="xml"><a class="salsah-link" '
+        'href="IRI:res_A_18:IRI">res_A_18</a></text><text permissions="prop-default" encoding="xml"><a '
+        'class="salsah-link" href="IRI:res_B_18:IRI">res_B_18</a></text></text-prop></resource>'
+    )
+    res_resptr, res_xml = _get_all_links_from_one_resource(test_ele)
+    expected_xml = [{"res_A_18"}, {"res_B_18"}]
+    assert not res_resptr
+    assert expected_xml == unordered(res_xml)
+
+
+def test_extract_id_one_text_with_one_id() -> None:
+    test_ele = etree.fromstring(
+        '<text permissions="prop-default" encoding="xml"><a class="salsah-link" '
+        'href="IRI:res_A_11:IRI">res_A_11</a></text>'
+    )
+    res = _extract_ids_from_one_text_value(test_ele)
+    expected = {"res_A_11"}
+    assert expected == res
+
+
+def test_extract_id_one_text_with_iri() -> None:
+    test_ele = etree.fromstring(
+        '<text permissions="prop-default" encoding="xml"><a class="salsah-link" '
+        'href="http://rdfh.ch/0801/RDE7_KU1STuDhHnGr5uu0g">res_A_11</a></text>'
+    )
+    res = _extract_ids_from_one_text_value(test_ele)
+    assert res == set()
+
+
+def test_extract_id_one_text_with_several_id() -> None:
+    test_ele = etree.fromstring(
+        '<text permissions="prop-default" encoding="xml"><a class="salsah-link" '
+        'href="IRI:res_A_11:IRI">res_A_11</a><a class="salsah-link" href="IRI:res_B_11:IRI">res_A_11</a><a '
+        'class="salsah-link" href="IRI:res_B_11:IRI">res_A_11</a></text>'
+    )
+    res = _extract_ids_from_one_text_value(test_ele)
+    expected = {"res_A_11", "res_B_11"}
```

Comment:
Is it worth having a variable for something as short as this?

(several more occurrences are further down, I won't comment them any more)